blog/07-01-23-how-to-build-car-marketplace-dapp-with-react/how-to-build-car-marketplace-dapp-with-react.md

```diff
@@ -0,0 +1,1195 @@
+    
+    address internal cUsdTokenAddress = 0x874069Fa1Eb16D44d622F2e0Ca25eeA172369bC1;
+
+    struct Car{
+        address payable owner;
+        string carName;
+        string carDescription;
+        string carImage; 
+        uint price;
+        bool isUsed; 
+        bool isBought;
+    }
+    
+    mapping (uint => Car) internal cars;
+   
+    function setCar(
+        string memory _carName,
+        string memory _carDescription,
+        string memory _carimage,
+        bool _isUsed,
+        uint _price
+    )public {
+        cars[carLength] = Car(
+              payable(msg.sender),
+              _carName,
+              _carDescription,
+              _carimage,
+              _price,
+              _isUsed,
+              false
+        );
+        carLength++;
+    }
+    
+    // this function reads a particular car data
+    function getCar (uint _index) public view returns (
+        address payable,
+        string memory,
+        string memory,
+        string memory,
+        uint,
+        bool,
+        bool
+    ) {
+        Car storage car = cars[_index];
+        return(
+          car.owner,
+          car.carName,
+          car.carDescription,
+          car.carImage,
+          car.price,
+          car.isUsed,
+          car.isBought
+        );
+    }
+    
+    // function to buy a car
+    function buyCar(uint _index) public  payable {
+        require(cars[_index].isBought == false,"Not for Sale");
+        require(
+          IERC20Token(cUsdTokenAddress).transferFrom(
+            msg.sender,
+            cars[_index].owner,
+            cars[_index].price
+          ),
+          "This transaction could not be performed"
+        );
+        
+        // change owner
+        cars[_index].owner = payable(msg.sender);
+        // change the sale  status
+        cars[_index].isBought = true;
+        
+    }
+
+    // function for user to sell his own car
+    function sellCar(uint _index) public {
+        require(msg.sender == cars[_index].owner,"Accessible only to the owner");
+        cars[_index].isBought = false;
+    }
+    
+    // function to get the length of the car array
+    function getCarLength() public view returns (uint) {
+        return (carLength);
+    }
+
+} 
+```
+
+## Contract Deployment
+
+To deploy the contract, we would need:
+1. [CeloExtensionWallet]((https://chrome.google.com/webstore/detail/celoextensionwallet/kkilomkmpmkbdnfelcpgckmpcaemjcdh?hl=en))
+2. [ Celo Faucet](https://celo.org/developers/faucet) 
+3. Celo Remix Plugin
+
+Download the Celo Extension Wallet from the Google chrome store using the link above. After doing that, create a wallet, store your key phrase in a very safe place to avoid permanently losing your funds.
+
+After downloading and creating your wallet, you will need to fund it using the Celo Faucet. Copy the address to your wallet, click the link to the faucet above and the paste the address into the text field and confirm.
```

Comment:
381 line 2 -correct line->faucet above, paste the address into the text field, and confirm.